unfetter-discover/unfetter-ui
# Issue 1145/bl ap chart

Fixes unfetter-discover/unfetter#1145
Fixes unfetter-discover/unfetter#1216

This PR fixes the AP chart, as well as the weightings for protect, detect, and respond, in the baseline summary.  The latter are essentially sums of the PDR scores, weighted accordingly (i.e. significant = 0.9, moderate = 0.6, limited = 0.3).  Lastly, the % complete reflects the weightings entered by the user against the complete number of attack patterns.

### To test:
1. Pull this branch
2. Create a baseline and leave default weightings in place
3. Once saved, note the % complete - it should be a very small percentage, since the goal is to have all attack patterns with PDR scores entered for all capabilities in the baseline (i.e. a total of 729 scores for each capability if using the NTCTF).
4. Note the AP weighted/unweighted chart - it should indicate a very small amount of complete AP weighted
5. Note the Protect, Detect, and Respond weightings - they should just be a summary of the scores with relevant weights applied to them (see above)
6. Make some changes to the PDR scores of existing capabilities and note changes in the PDR weightings on the summary page
7. Add some attack patterns and/or capabilities, and the AP weightings and % complete should update accordingly
